applications/visualizer/backend/ingestion/validator.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from enum import IntEnum, StrEnum
+from typing import Dict, List, Literal, Optional, Tuple
+
+from pydantic import BaseModel, Field, RootModel, model_validator
+
+
+class Neuron(BaseModel):
+    inhead: bool  # int used as bool, is the neuron part of the head or not
+    name: str  # name of the neuron, can be same as classes, or L or R of classes
+    emb: bool  # int used as bool
+    nt: str  # neurotransmitter type
+    intail: bool  # int used as bool
+    classes: str  # general name of the neuron
+    typ: str  # type of the neuron
+
+
+class DatasetType(StrEnum):
+    COMPLETE = "complete"
+    HEAD = "head"
+    TAIL = "tail"
+
+
+class Axe(BaseModel):
+    face: str
+    axisIndex: int
+    axisTransform: int
+
+
+class Dataset(BaseModel):
+    id: str
+    name: str
+    type: DatasetType
+    time: float  # TODO: should add validation gte than 0?
+    visualTime: float  # TODO: should add validation gte than 0?
+    description: str
+    axes: Optional[List[Axe]] = Field(
+        default=None, description="different axes and their representation"
+    )
+
+
+class ConnectionType(IntEnum):
+    ELECTRICAL = 0
+    CHEMICAL = 2
+
+
+class Connection(BaseModel):
+    ids: List[int] = Field(
```

Comment:
Initially I was doing the type annotations like you suggest. My thought process to go back on it was to support as many python interpreters as I could. Since this will run on the user's machines, I didn't want users to be concern with which interpreter to use and such.

Reply:
But I agree with you on this, will change it.

Reply:
That's a good point about the version support. The support for `__gettitem__` on types is here since Python 3.9 though and Python 3.8 will be in end of life in October, I think we can assume we only support Python versions that are still officially supported :D